security/src/main/java/de/evoila/cf/security/keystore/KeyStoreHandler.java

```diff
@@ -0,0 +1,117 @@
+package de.evoila.cf.security.keystore;
+
+import org.bouncycastle.util.io.pem.PemObject;
+import org.bouncycastle.util.io.pem.PemReader;
+import org.springframework.stereotype.Service;
+import sun.security.util.DerInputStream;
+import sun.security.util.DerValue;
+
+import javax.naming.ConfigurationException;
+import java.io.ByteArrayInputStream;
+import java.io.IOException;
+import java.io.StringReader;
+import java.math.BigInteger;
+import java.security.GeneralSecurityException;
+import java.security.KeyFactory;
+import java.security.KeyStore;
+import java.security.PrivateKey;
+import java.security.cert.Certificate;
+import java.security.cert.CertificateFactory;
+import java.security.spec.PKCS8EncodedKeySpec;
+import java.security.spec.RSAPrivateCrtKeySpec;
+import java.util.Base64;
+
+/**
+ * Created by reneschollmeyer, evoila on 21.11.18.
+ */
+@Service
+public class KeyStoreHandler {
+
+    public KeyStore getKeyStore(String unformattedClientCertificate, String unformattedPrivateKey, String unformattedCaCertificate, String keyStorePassword) throws ConfigurationException {
+        try {
+            Certificate clientCertificate = loadCertificate(unformattedClientCertificate);
+            PrivateKey privateKey = loadPrivateKey(unformattedPrivateKey);
+            Certificate caCertificate = loadCertificate(unformattedCaCertificate);
+
+            KeyStore keyStore = KeyStore.getInstance("JKS");
+            keyStore.load(null, null);
```

Comment:
Why is this null, null?

Reply:
This is copy pasted, im not sure about this

Reply:
Please check

Reply:
The load method is actually for loading a given keystore. Parameters are 

  * @param stream: the input stream from which the keystore is loaded, or {@code null}
  * @param password: the password used to check the integrity of the keystore, the password used to unlock the keystore or {@code null}

However, when you (like in our case) create a new keystore, this method has to be called nevertheless because it sets an important flag "initialized" to true. In this case you don't need any parameters though, that's why both of them are null.

Reply:
ok, good for me.